### PR TITLE
Moves all native shared libraries to a common shared directory between java and c++

### PR DESCRIPTION
**Pull request by ThadHouse**
Greatly reduces file duplication.

### CODE REVIEW

File: edu.wpi.first.wpilib.plugins.core/pom.xml [hunk 1]
```diff
@@ -122,6 +187,65 @@
                                     <outputDirectory>${tools-zip}</outputDirectory>
                                     <destFileName>OutlineViewer.jar</destFileName>
                                 </artifactItem>
+
+
```
Comment (PeterJohnson) [inline edu.wpi.first.wpilib.plugins.core/pom.xml]: Extra blank line (or two?)

File: edu.wpi.first.wpilib.plugins.core/pom.xml [hunk 2]
```diff
@@ -122,6 +187,65 @@
...
+                                </artifactItem>
+                                <artifactItem>
+                                    <groupId>org.opencv</groupId>
+                                    <artifactId>opencv-cpp</artifactId>
+                                    <classifier>linuxathena</classifier>
+                                    <type>zip</type>
+                                    <outputDirectory>${project.build.directory}</outputDirectory>
+                                    <destFileName>opencv.jar</destFileName>
+                                </artifactItem>
+
```
Comment (PeterJohnson) [inline edu.wpi.first.wpilib.plugins.core/pom.xml]: Could probably remove this blank line too

**Comment by ThadHouse** [comment]
Fixed all of these

File: edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/build.properties [hunk 3]
```diff
@@ -10,7 +10,8 @@ command.dir=/home/lvuser/
 
 # Libraries to use
 wpilib=${user.home}/wpilib/cpp/${cpp-version}
-wpilib.lib=${wpilib}/lib/linux/athena/shared
+wpilib.common=${user.home}/wpilib/common
```
Comment (PeterJohnson) [inline edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/build.properties]: This fails.  There should be a version in here ala Java?

**Comment by ThadHouse** [comment]
@frcjenkins  retest this please

**Comment by ThadHouse** [comment]
Closes #130

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**